Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -33,17 +33,21 @@ public class IotHubConnectionStringManager : IIotHubConnectionStringManager
         private readonly ILogger log;
         private readonly IDiagnosticsLogger diagnosticsLogger;
         private readonly IStorageRecords mainStorage;
-
+        private readonly IRegistryManager registryManager;
+        
         public IotHubConnectionStringManager(
             IServicesConfig config,
             IFactory factory,
             IDiagnosticsLogger diagnosticsLogger,
+            IRegistryManager registryManager,
             ILogger logger)
         {
             this.config = config;
             this.mainStorage = factory.Resolve<IStorageRecords>().Init(config.MainStorage);
             this.log = logger;
             this.diagnosticsLogger = diagnosticsLogger;
+            this.registryManager = registryManager;
+            this.registryManager.Init(this.config.IoTHubConnString);
```

Comment:
what happens if we deploy the solution without a hub and the conn string in the config is empty? what if we have simulations using different conn strings?

Reply:
moving initialization of registry manager into the function "TestIoTHubReadPermissionsAsync". This way, the scenarios of connection string being empty or invalid would have been handled even before the function gets called.